### ISSUE TITLE
1.5: HeadingPermalinkProcessor::getChildText() strips out AbstractStringContainer from link

### ISSUE THREAD

**Issue by Ambient-Impact** [thephpleague/commonmark#615]
**Description**  
I know that [```HeadingPermalinkProcessor::getChildText()```](https://github.com/thephpleague/commonmark/blob/cf52b9ffceb8e2e37aa4fb9503c9753721ae1bc7/src/Extension/HeadingPermalink/HeadingPermalinkProcessor.php#L122) is going to be removed in 2.0, but I'm stuck with 1.5 for the foreseeable future. I assumed any class that extends ```AbstractStringContainer``` would have its text contents included in the permalink, but it seems that ```instanceof Text``` and ```instanceof Code``` are hard coded. Is there a reason for this, or can it be expanded to ```instanceof AbstractStringContainer```?

**Example**  
I'm helping bring [8fold/commonmark-abbreviations](https://github.com/8fold/commonmark-abbreviations) closer in line with CommonMark core (see 8fold/commonmark-abbreviations#8), and when an ```Abbreviation``` is present in the heading text, even with the pull request to extend ```AbstractStringContainer```, the text content is completely ignored by the heading permalink.

**Comment by Ambient-Impact** [comment]
Upon poking around, it seems like this issue can be worked around with a custom text normalizer:

```php
class YourNormalizer implements TextNormalizerInterface {

  /**
   * {@inheritdoc}
   */
  public function normalize(string $text, $context = null): string {
    if (
      \is_object($context) &&
      \method_exists($context, 'getStringContent') &&
      !empty($context->getStringContent())
    ) {
      $text = $context->getStringContent();
    }

    // Rest of normalizer processing goes here.

  }

}

**Comment by colinodell** [comment]
> I assumed any class that extends AbstractStringContainer would have its text contents included in the permalink, but it seems that `instanceof Text` and `instanceof Code` are hard coded. Is there a reason for this, or can it be expanded to `instanceof AbstractStringContainer`?

I honestly can't recall why the other node types were excluded.

Given that 2.0 allows other types including `AbstractStringContainer`, I don't see any obvious reasons why we can't allow this in 1.x too.

Would you be okay if we changed this behavior in 1.6.1? Or do you really need this in 1.5?

**Comment by Ambient-Impact** [comment]
Given the workaround I posted, it's no rush and fine in 1.6.1. Thanks! 🤓

**Comment by colinodell** [comment]
Fixed via https://github.com/thephpleague/commonmark/commit/a70a5865200784144bf7ae1086f21633094badc9 and released in [1.6.1](https://github.com/thephpleague/commonmark/releases/tag/1.6.1).

Thanks for reporting this!

**Comment by Ambient-Impact** [comment]
Awesome, thanks!

[![Excited wiggling](https://media2.giphy.com/media/Ir6QkDxjztyZa/giphy.gif?cid=790b76117b0994b91aa581d5aec1b3616031f5cdcc743418&rid=giphy.gif&ct=g)](https://giphy.com/gifs/dodger-Ir6QkDxjztyZa)
